### PR TITLE
Fix snapshot probabilities extension

### LINKED ISSUE #304: Probabilities snapshot extension
### What is the expected behavior?

This will be used by various methods of the QasmSimulator to return snapshots of the measurement outcome probabilities of the specified qubits.

```python
class SnapshotProbabilities(Snapshot):
    __init__(self, label, num_qubits, variance=False):
        if variance:
            super(label, "probabilities_with_variance", num_qubits, 0, None)
        else:
            super(label, "probabilities", num_qubits, 0, None)


# Monkey patch function
def snapshot_probabilities(self, label, qubits=None, variance=False):
     # If qubits is None snapshot all qubits.

circuit.snapshot_probabilities = snapshot_probabilities
```

### PR DESCRIPTION
**Pull request by chriseclectic**
### Summary

Fixes snapshot probabilities extension to work on any number of qubits.
Adds tests for probability snapshots

Closes #304 

~Blocked by #374~

### CODE REVIEW

File: test/terra/backends/qasm_simulator/qasm_snapshot.py [hunk 1]
```diff
@@ -484,3 +486,84 @@ def test_snapshot_density_matrix_post_measure_nondet(self):
...
+    def probabilitiy_snapshots(data, labels):
+        """Format snapshots as nested dicts"""
+        # Check snapshot entry exists in data
+        output = {}
+        for label in labels:
+            snaps = data.get("snapshots", {}).get("probabilities", {}).get(label, [])
+            # Convert list into dict
+            inner = {}
+            for snap_dict in snaps:
+                inner[snap_dict['memory']] = snap_dict['value']
```
Comment (atilag) [inline test/terra/backends/qasm_simulator/qasm_snapshot.py]: Nit: I was wondering how to write this using comprehesion dictionaries, but I don't like the result, what do you think?
```python
inner = { k: v for m,k in snaps.items() if m == "memory" for m,v in snaps.items() if  m == "value" }
```

Comment (atilag) [reply test/terra/backends/qasm_simulator/qasm_snapshot.py]: hahahaha ok, way better (thanks @delapuente)
```python
inner = { snap_dict['memory']: snap_dict['value'] for snap_dict in snaps }
```

Comment (atilag) [reply test/terra/backends/qasm_simulator/qasm_snapshot.py]: ... there are days, you know..

File: qiskit/providers/aer/extensions/snapshot_probabilities.py [hunk 2]
```diff
@@ -18,47 +18,50 @@
...
 
     return self.append(
-        SnapshotProbabilites(label,
-                             num_qubits=len(snapshot_register),
-                             variance=variance,
-                             params=params), snapshot_register)
+        SnapshotProbabilities(label,
+                              num_qubits=len(snapshot_register),
+                              variance=variance),
+        snapshot_register)
```
[the remark this answers is not in the document]
Comment (chriseclectic) [reply qiskit/providers/aer/extensions/snapshot_probabilities.py]: If yapf was happy I guess so

**Pull request merged**